src/Rings/mpoly-ideals.jl

```diff
@@ -1683,10 +1705,23 @@ false
 ```
 """
 @attr Bool function is_prime(I::MPolyIdeal)
+  is_known(primary_decomposition, I) && is_one(length(primary_decomposition(I))) && return is_subset(first(primary_decomposition(I))[2], I)
```

Comment:
OK reversing my earlier question about `is_known(I::MPolyIdeal, ::typeof(is_prime))`, should this perhaps be improved (?) to
```suggestion
  if is_known(primary_decomposition, I)
    length(primary_decomposition(I)) > 1 && return false
    return is_subset(first(primary_decomposition(I))[2], I)
  end
```

Reply:
Sorry, didn't see it in time; got lost in my inbox. But this is only about style, right? No mathematical error?